thingsboard_gateway/connectors/modbus_async/slave.py

```diff
@@ -0,0 +1,233 @@
+            self.connection_attempt_count = 0
+            self.last_connection_attempt_time = cur_time
+            return True
+        else:
+            return False
+
+    def set_master_connection(self, master_connection):
+        self.master = master_connection
+        self.available_functions = self.__get_available_functions()
+
+    def __get_available_functions(self):
+        return {
+            1: self.master.read_coils,
+            2: self.master.read_discrete_inputs,
+            3: self.master.read_holding_registers,
+            4: self.master.read_input_registers,
+            5: self.master.write_coil,
+            6: self.master.write_register,
+            15: self.master.write_coils,
+            16: self.master.write_registers,
+        }
+
+    async def read(self, function_code, address, objects_count):
+        self._log.debug('Read %s registers from address %s with function code %s', objects_count, address,
+                        function_code)
+
+        result = await self.__read(function_code, address, objects_count)
+
+        StatisticsService.count_connector_message(self.name, stat_parameter_name='connectorMsgsReceived')
+        StatisticsService.count_connector_bytes(self.name, result, stat_parameter_name='connectorBytesReceived')
+
+        return result
+
+    async def __read(self, function_code, address, objects_count):
+        result = None
+
+        try:
+            result = await self.available_functions[function_code](address=address, count=objects_count, slave=self.unit_id)
+        except KeyError:
+            self._log.error('Unknown Modbus function with code: %s', function_code)
+
+        self._log.debug("Read with result: %s", str(result))
+        return result
+
+    async def write(self, function_code, address, value):
+        self._log.debug('Write %s value to address %s with function code %s', value, address, function_code)
+        result = await self.__write(function_code, address, value)
+
+        StatisticsService.count_connector_message(self.name, stat_parameter_name='connectorMsgsReceived')
+        StatisticsService.count_connector_bytes(self.name, result, stat_parameter_name='connectorBytesReceived')
+
+        return result
+
+    async def __write(self, function_code, address, value):
+        result = None
+
+        if function_code in (5, 6):
+            result = await self.available_functions[function_code](address=address, value=value, slave=self.unit_id)
+        elif function_code in (15, 16):
+            result = await self.available_functions[function_code](address=address, values=value, slave=self.unit_id)
+        else:
+            self._log.error("Unknown Modbus function with code: %s", function_code)
+
+        self._log.debug("Write with result: %s", str(result))
+        return result
+
+    @staticmethod
+    def set_connect_attempt_time_ms(connect_attempt_time_ms):
+        return connect_attempt_time_ms if connect_attempt_time_ms >= 500 else 500
```

Comment:
Do this and 2 following methods are required? You can check these points during __init__.